staging/src/k8s.io/component-base/tracing/tracing.go

```diff
@@ -43,6 +43,20 @@ func Start(ctx context.Context, name string, attributes ...attribute.KeyValue) (
 	}
 }
 
+// StartNewRootSpan create new root span using both OpenTelemetry, and the k8s.io/utils/trace package.
+// Incoming context is not required to contain span parameters
+func StartNewRootSpan(ctx context.Context, tp trace.TracerProvider, name string, attributes ...attribute.KeyValue) (context.Context, *Span) {
```

Comment:
As another note, the tracer should be created once for each library that is instrumented with tracing, rather than recreated each time a span is made.

Reply:
>component-base/tracing is for components that are migrating from "util" tracing to opentelemetry-based tracing

Is component-base/tracing expected to be removed any time soon? If not I'd rather use it for consistency. Root span don't make sense in `util` tracing though, agreed, removed these lines from `StartNewRootSpan`

> the tracer should be created once for each library that is instrumented with tracing, rather than recreated each time a span is made

TracerProvider is caching created tracers, so a tracer is created just once.

Reply:
I would expect it to be removed once all components using it are migrated to OpenTelemetry, and after a deprecation period.  That probably isn't soon, but we wouldn't want users to take a dependency on the log-based traces produced in the meantime.

It is cached, but it is still unnecessary to reconstruct the scope each time (e.g. https://github.com/open-telemetry/opentelemetry-go-contrib/pull/2835)

Reply:
So I think I should
* avoid using `tracing` entirely and use native OpenTelemetry calls
* pass `Tracer` instead of `TracerProvider` so that would not be reconstructed every call

Do that sound right?

Reply:
> avoid using tracing entirely and use native OpenTelemetry calls

Correct.

> pass Tracer instead of TracerProvider so that would not be reconstructed every call

The Tracer should point to the library that is instrumented, and should be constructed once.  Probably the best approach would be to have one Tracer for each package (e.g. for `kubernetes.io/kubernetes/pkg/kubelet`, another for `kubernetes.io/kubernetes/pkg/kubelet/kuberuntime`, etc.)

Reply:
Done, not touching `tracing` and creating 3 Tracers: kubelet, kubelet/kuberuntime and kubelet/images